File: sohee-jung/week4/백준_silver_바이러스.java
import java.io.*;
import java.util.ArrayList;
import java.util.List;
import java.util.StringTokenizer;

public class Main {
    static List<Integer>[] computerGraph;
    static boolean[] visited;

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));

        int computerNumbers = Integer.parseInt(br.readLine());
        int computerLinks = Integer.parseInt(br.readLine());

        computerGraph = new ArrayList[computerNumbers+1];
        visited = new boolean[computerNumbers+1];

        for (int i = 0; i <= computerNumbers; i++) {
            computerGraph[i] = new ArrayList<>();
        }

        for (int i = 0; i < computerLinks; i++) {
            StringTokenizer st = new StringTokenizer(br.readLine());
            int firstNum = Integer.parseInt(st.nextToken());
            int secondNum = Integer.parseInt(st.nextToken());

            computerGraph[firstNum].add(secondNum);
            computerGraph[secondNum].add(firstNum);
        }

        int answer = DFS(1)-1;
        bw.write(answer + "\n");

        bw.flush();
        br.close();
        bw.close();
    }

    static int DFS(int node){
        visited[node] = true;
        int count = 1;
        for (int childNode: computerGraph[node]){
            if (!visited[childNode]){
                count += DFS(childNode);
            }
        }
        return count;
    }
}

File: minseo-song/week4/백준_silver_바이러스.kt
package week4

import java.io.BufferedReader
import java.io.BufferedWriter
import java.io.InputStreamReader
import java.io.OutputStreamWriter
import java.util.StringTokenizer

fun main() {
    val bufferedReader = BufferedReader(InputStreamReader(System.`in`))
    val bufferedWriter = BufferedWriter(OutputStreamWriter(System.out))

    val n = bufferedReader.readLine().toInt()
    val m = bufferedReader.readLine().toInt()

    val graph = Array(n + 1) { mutableListOf<Int>() }
    val visited = BooleanArray(n + 1)

    repeat(m) {
        val edge = StringTokenizer(bufferedReader.readLine())
        val x = edge.nextToken().toInt()
        val y = edge.nextToken().toInt()

        graph[x].add(y)
        graph[y].add(x)
    }

    var count = 0

    fun dfs(node: Int){
        visited[node] = true
        for (next in graph[node]){
            if (!visited[next]){
                count++
                dfs(next)
            }
        }
    }

    dfs(1)

    bufferedWriter.write(count.toString())
    bufferedWriter.close()
}